### PR TITLE
create iam role for github runners to access aws secrets manager

### PR DESCRIPTION
**Pull request by angeloskyratzakos**
#### Summary
create iam role for github runners to access aws secrets manager

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-8657

#### Release Note

```release-note
create iam role for github runners to access aws secrets manager
```

### CODE REVIEW

**Review by andrleite — DISMISSED** [review]

File: aws/github-runner/iam.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  name               = "GitHubSecretsAccessRole"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
```
Comment (toninis) [inline aws/github-runner/iam.tf:11]: The principal needs to be the role of the service account

File: aws/github-runner/iam.tf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  name = "GitHubSecretsAccessRole"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          AWS = var.service_account_role_arn
+        }
+        Action = "sts:AssumeRole"
```
Comment (toninis) [inline aws/github-runner/iam.tf:11]:
```suggestion
          AWS = var.service_account_role_arn
        }
        Action = [
                "sts:TagSession",
                "sts:AssumeRole"
          ]
```

**Review by toninis — CHANGES_REQUESTED** [review]
It needs some extra perms to work 😄

**Review by toninis — APPROVED** [review]

**Review by stafot — APPROVED** [review]
lgtm

**Pull request merged**